### PR TITLE
Removes Accidental IC in OOC

### PR DESCRIPTION
**Pull request by Incoming5643**
OOC will catch messages that start with "." ";", ":", "#" (the say channel modifiers) or "say" itself.

When you type something that looks like IC in OOC you'll get a pop up that will let you actually say it if that's what you meant to do. It defaults to no so real accidental IC OOC can mash out of the command in case what they said was something like ";help I'm being killed!"

:cl: incoming5643
rscadd: OOC will now take steps to try and prevent accidental IC from people who mistakenly use OOC instead of say. This doesn't catch all mistakes, so please don't test it.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
GET THE FUCK OUT

**Comment by Anonus** [comment]
# is also a say channel modifier, not that I've ever seen someone use it in ooc accidentally.

**Comment by MrStonedOne** [comment]
Don't make it scrub it out, have it have a link the player can click to say it anyways or a pop up

having to add a space is bullshit

**Comment by PKPenguin321** [comment]
add "say" to the list

**Comment by Anonmare** [comment]
>No more tik-tok ick ock
Why do you hate fun?

**Comment by optimumtact** [comment]
>(if you want to make a fake IC in OOC joke you still can by adding a space before the message)

**Comment by Incoming5643** [comment]
Alright I've solved the problem in a way that lets you shitpost more effectively, happy?

File: code/game/verbs/ooc.dm [hunk 1]
```diff
@@ -14,6 +14,10 @@
 	msg = copytext(sanitize(msg), 1, MAX_MESSAGE_LEN)
 	if(!msg)	return
 
+	if((copytext(msg, 1, 2) in list(".",";",":","#")) || (lowertext(copytext(msg, 1, 4)) == "say"))
+		if(alert("Your message \"[msg]\" looks like it was meant for in game communication, say it in OOC?", "Meant for OOC?", "No", "Yes") != "Yes")
```
Comment (Iamgoofball) [inline code/game/verbs/ooc.dm]: idea:
if you hit no, make it say() like you intended it to

Comment (Incoming5643) [inline code/game/verbs/ooc.dm]: OOC is the same for everyone, but say is pretty dependent on what you are. I'm afraid.

**Comment by MrStonedOne** [comment]
much better.

**Comment by PKPenguin321** [comment]
still needs "say" added to the list
please do this before i have to make another pr to do it once this is merged

**Comment by Incoming5643** [comment]
I put in say, scroll right

**Comment by PKPenguin321** [comment]
aha cool i'm blind
good job :+1:

**Comment by MrStonedOne** [comment]
you could add asay and dsay if you really want to get jiggy with it.

**Comment by Iamgoofball** [comment]
@MrStonedOne wouldn't the say one pick those up already

**Comment by RemieRichards** [comment]
@Iamgoofball it's a string check for "say", not checking whether "say" occurs in the string, primarily because of stuff like using say as an actual word "What did he say?" I assume.

**Comment by Dorsisdwarf** [comment]
worthy

**Comment by WJohn** [comment]
:+1: x :100:

**Comment by Robustin** [comment]
Yea I need this in my life.

Getting banned while im playing a head/antag for some mid-level IC shit by accident ain't cool.

**Comment by Aranclanos** [comment]
>popup windows

jesus christ how horrifying

**Comment by RemieRichards** [comment]
@Incoming5643 asay and dsay pls.

**Comment by Incoming5643** [comment]
now checks for "say" within first four letters of message. This should catch "dsay" and "asay" as well as a few big mistypes like ";say". There aren't many words that <a href="http://www.wordfind.com/starts-with/say/">start with say</a> to false positive and there are NO words that have say as the second through forth letters.

Also moves the check below ooc mutes and such

**Pull request merged**